torchrl/modules/models/multiagent.py

```diff
@@ -229,6 +229,141 @@ def forward(self, *inputs: Tuple[torch.Tensor]) -> torch.Tensor:
+        >>> batch = (3,2)
+        >>> n_agents = 7
+        >>> x, y, channels = 100, 100, 3
+        >>> obs = torch.randn(*batch, n_agents, x, y, channels)
+        >>> cnn = MultiAgentConvNet(
+        >>>     n_agents,
+        >>>     centralised = True,
+        >>>     share_params = True
+        >>> )
+        >>> print(cnn)
+        MultiAgentConvNet(
+            (agent_networks): ModuleList(
+                (0): ConvNet(
+                (0): LazyConv2d(0, 32, kernel_size=(5, 5), stride=(2, 2))
+                (1): ELU(alpha=1.0)
+                (2): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (3): ELU(alpha=1.0)
+                (4): Conv2d(32, 32, kernel_size=(5, 5), stride=(2, 2))
+                (5): ELU(alpha=1.0)
+                (6): SquashDims()
+                )
+            )
+        )
+        >>> print(cnn(obs).shape)
+        torch.Size([3, 2, 7, 2592]) - where the final dimension is determiend by the input x,
+
+    """
+    def __init__(self,
+                 n_agents: int,
+                 centralised: bool,
+                 share_params: bool,
+                 device: Optional[DEVICE_TYPING] = None,
+                 num_cells: Optional[Sequence[int]] = None,
+                 kernel_sizes: Union[Sequence[Union[int, Sequence[int]]], int] = 5,
+                 strides: Union[Sequence, int] = 2,
+                 paddings: Union[Sequence, int] = 0,
+                 activation_class: Type[nn.Module] = nn.ELU,   
+                 **kwargs):
+        super().__init__()
+
+        self.n_agents = n_agents
+        self.centralised = centralised
+        self.share_params = share_params
+
+        self.agent_networks = nn.ModuleList(
+            [
+                ConvNet(
+                    num_cells=num_cells,
+                    kernel_sizes=kernel_sizes,
+                    strides=strides,
+                    paddings=paddings,
+                    activation_class=activation_class,
+                    device=device,
+                    **kwargs
+                )
+                for _ in range(self.n_agents if not self.share_params else 1)
+            ]
+        )
+
+
+    def forward(self, *inputs: Tuple[torch.Tensor]):
+        if len(inputs) > 1:
+            inputs = torch.cat([*inputs], -1)
+        else:
+            inputs = inputs[0]
+
+        assert len(inputs.shape) >= 4, \
+f"""Multi-agent network expects" (*batch_size, agent_index, x, y, channels)"""
+        assert inputs.shape[-4] == self.n_agents, \
+f"""Multi-agent network expects {self.n_agents} but got {inputs.shape[-4]}"""
+        # If the model is centralized, agents have full observability
+        if self.centralised:
+            shape = (*inputs.shape[:-4], inputs.shape[-3], inputs.shape[-2], self.n_agents * inputs.shape[-1])
+            inputs = torch.reshape(inputs, shape)
+        # Conv2D expects (c, H, W)
+        inputs = torch.transpose(inputs, -1, -3)
```

Comment:
We want to keep the exact same behaviour of torchrl.models.ConvNet. Users that have (x,y,c) input should do this in a transform (ToTensorImage).

Could we refactor the class to take by default (c, H, W)?

Reply:
Refactored class and doc to expect (c, H, W)